File: insertionSort.py

def insertionSort(arr, n):
    for i in range(1, n):
        curr = arr[i]
        prev = i - 1
        while prev >= 0 and arr[prev] > curr:
            arr[prev + 1] = arr[prev]
            prev -= 1
        arr[prev + 1] = curr

def printArray(arr, n):
    for i in arr:
        print(i, end=" ")
    print()

n = 5;
arr = [4,1,5,2,3]
insertionSort(arr, n)
printArray(arr, n)

File: insertionSort.cpp
#include <iostream>
using namespace std;

void selectionSort(int arr[], int n) {
    for(int i = 1; i < n; i++) {
        int current = arr[i];
        int prev = i - 1;
        while(prev >= 0 && arr[prev] > current) {
            arr[prev + 1] = arr[prev];
            prev--;
        }
        arr[prev + 1] = current;
    }
}

void printArray(int arr[], int n) {
    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    int n = 5;
    int arr[] = {4,1,5,2,3};

    selectionSort(arr, n);
    printArray(arr, n);

    return 0;
}